### ISSUE TITLE
Emit `extern`?

### ISSUE THREAD

**Issue by roman01la** [carp-lang/Carp#294]
Hello! 👋 I'm experimenting with compiling Carp to WASM (it works!), could we have a syntax to emit `extern` annotation somehow? This way you export functions/vars into JS world from compiled wasm code. For now I manually edit emited C code :(

Would be great if someone could drop a quick guide on where too look in the source, so I could try to implement this and learn Carp's internals.

Thanks for amazing work!

**Comment by hellerve** [comment]
Thanks for your interest in Carp!

Late in 2017, I wrote a—now a little outdated—[overview of how the compiler works](https://blog.veitheller.de/The_Carp_Compiler_(as_of_2017).html) on my blog. Short answer: the C is generated in `src/Emit.hs`.

I also experimented with WASM for a bit before (see #178). I dropped it, because I ddidn’t end up having time for it; it would be great if you find the time and interest to actually power through!
